src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,291 @@
+                    state = NM_CONNECTION_ACTIVATED
+
+            else:
+                self.logger.error('Active connection for connection %s is no longer valid.',
+                                  connection_id)
+
+        return state
+
+    def _get_active_connection(self, connection_id):
+        """Returns the active connection object associated with the given network id.
+
+        connection_id: The displayed name of the connection in NetworkManager.
+        """
+        active_connection = None
+
+        active_connections = NetworkManager.NetworkManager.ActiveConnections
+        self._run_proxy_call(active_connections)
+
+        for available_active_connection in active_connections:
+            if available_active_connection.Id == connection_id:
+                self.logger.debug('Found active connection for connection %s.',
+                                  connection_id)
+                active_connection = available_active_connection
+
+        return active_connection
+
+    def _wait_for_connection(self, connection):
+        """Wait for the configured number of seconds for an active connection to be ready.
+        return True if it connects in that time, False otherwise.
+
+        connection: The NetworkManager.Connection object to watch for connectivity.
+        """
+        success = False
+        give_up = False
+        connection_id = connection.GetSettings()['connection']['id']
+
+        self.logger.debug('Waiting for connection %s...', connection_id)
+        time_to_give_up = time.time() + self.network_activation_timeout
+        while (success is False and give_up is False):
+
+            connection_state = self._get_connection_state(connection_id)
+
+            if connection_state is NM_CONNECTION_ACTIVATED:
+                self.logger.debug('Connection %s successful.', connection_id)
+                success = True
+
+            elif connection_state is NM_CONNECTION_DISCONNECTED:
+                self.logger.warning('Connection %s disconnected. Giving up.', connection_id)
+                give_up = True
+
+            elif time.time() > time_to_give_up:
+                self.logger.warning('Connection %s timed out. Giving up.', connection_id)
+                give_up = True
+
+            else:
+                time.sleep(NETWORKMANAGER_ACTIVATION_CHECK_DELAY)
+
+        return success
+
+    def _run_proxy_call(self, proxy_call):
+        """If proxy_call is callable, call it. This typically means that there was a
+        critical D-Bus error and an exception will most likely be raised.
+
+        proxy_call: The output of a NetworkManager D-Bus call.
+        """
+        # Sometimes, instead of raising exceptions or returning error states, NetworkManager
+        #   will return a function that makes a D-Bus call it assumes will fail. We still
+        #   want this error information, so here, we call it if we can. This method will
+        #   probably raise an exception if it is called.
```

Comment:
You could probably combine the inline comment with the class comment. Inline comments have more context, but not much context will be lost here because the method is only 2 lines. :)